src/jdk.javadoc/share/classes/jdk/javadoc/internal/tool/Messager.java

```
@@ -25,32 +25,87 @@
 
 /**
- * Utility for integrating with javadoc tools and for localization.
- * Handle resources, access to error and warning counts and
- * message formatting.
+ * Class for reporting diagnostics and other messages.
+ *
+ * The class leverages the javac support for reporting diagnostics, for stylistic consistency
+ * of diagnostic messages and to avoid code duplication.
+ *
+ * The class is a subtype of javac's Log, and is primarily a transducer between
+ * javadoc method signatures and the underlying javac methods. Within this class,
+ * the methods call down to a core {@code report} method which hands off to
+ * a similar method in the superclass ({@code Log.report}, which takes care
+ * of reporting the diagnostic (unless it has been suppressed), displaying
+ * the source line and a caret to indicate the position of the issue (if appropriate),
+ * counting errors and warnings, and so on.
+ *
+ * In general, the underlying javac layer is more powerful, whereas the javadoc methods are
+ * constrained by the public {@link jdk.javadoc.doclet.Doclet} API.
+ *
+ * In the underlying javac layer, the following abstractions are used:
+ * <ul>
+ *     <li>{@code DiagnosticType} -- error, warning, note, etc.
+ *     <li>{@code DiagnosticSource} -- a file object and a cache of its content
+ *     <li>{@code DiagnosticPosition} -- a tuple of values (start, pos, end) for the position of a diagnostic
+ *     <li>{@code DiagnosticFlag} -- additional flags related to the diagnostic
+ * </ul>
+ *
+ * The javadoc layer is defined by the methods on {@code Doclet.Reporter}, and by
+ * assorted methods defined in this class for use by the javadoc tool.
+ * The primary data types are:
+ * <ul>
+ *     <li>{@code Diagnostic.Kind} -- maps to {@code DiagnosticType} and {@code Set<DiagnosticFlag>}
+ *     <li>{@code Element} -- maps to {@code DiagnosticSource and {@code DiagnosticPosition}
+ *     <li>{@code DocTreePath} -- maps to {@code DiagnosticSource and {@code DiagnosticPosition}
```

Close `@code` tags.

---

oops

---

mildly worrying that "make docs" did not give errors  ... but I guess the error was masked by the subsequent `}` of the subsequent `{@code ...}`

---

> mildly worrying that "make docs" did not give errors

`-quiet` might be to blame.

---

> mildly worrying that "make docs" did not give errors

This should be currently detectable by DocLint: since `</ul>` is consumed by `@code`, the `ul` element is left unclosed.